client/main/lib/Client.ts

```diff
@@ -0,0 +1,38 @@
+import Datastore, { Crawler, Function, Table } from '@ulixee/datastore';
+import ILocationStringOrObject from "../interfaces/ILocationStringOrObject";
+import ClientForDatastore from "./ClientForDatastore";
+import ClientForFunction from './ClientForFunction';
+import ClientForTable from './ClientForTable';
+import ClientForCrawler from './ClientForCrawler';
+import ClientForRemote from './ClientForRemote';
+
+function Client(this: ClientForRemote, remote?: ILocationStringOrObject): void;
```

Comment:
Is this supposed to be returning void? Would think Typescript calling code will not know what to do with the return type.

Reply:
I cannot get this working without the `void`, but yes, you're correct, Typescript does not know what to do with the return type.

Reply:
I ripped all this entire file out and replace it with separate classes.